### PR TITLE
Move all login and account creation page labels to be above inputs

### PR DESCRIPTION
**Pull request by rafh**
There are a few inconsistencies within Gitea and this PR addresses one of them. This PR updates the sign-in page layout, including the register and openID tabs, to match the layout of the settings pages (/user/settings) for more consistency.

This PR updates the following routes:
`/user/login`
`/user/sign_up`
`/user/login/openid`
`/user/forgot_password`
`/user/link_account`
`/user/recover_account`

**Before**
Screenshot 2024-02-05 at 8 27 24 AM

**After**
Screenshot 2024-02-05 at 8 26 39 AM

This PR addresses a revert of the original PR due to this [comment](https://github.com/go-gitea/gitea/pull/28753#issuecomment-1956596817).

### CODE REVIEW

**Comment by KN4CK3R** [comment]
Could you have a look at the two factor page too? Last time I checked the content was not centered.
grafik

**Comment by silverwind** [comment]
Looks cleaner with that layout, I like it.

**Comment by zokkis** [comment]
Maybe remove the 2. title?
Because now there is the tab with "sign in" and directly under the tab is the title with "sign in"

File: templates/user/auth/captcha.tmpl [hunk 1]
```diff
@@ -1,11 +1,10 @@
 {{if .EnableCaptcha}}{{if eq .CaptchaType "image"}}
 	<div class="inline field">
-		<label>{{/* This is CAPTCHA field */}}</label>
 		{{.Captcha.CreateHTML}}
 	</div>
 	<div class="required inline field {{if .Err_Captcha}}error{{end}}">
 		<label for="captcha">{{ctx.Locale.Tr "captcha"}}</label>
-		<input id="captcha" name="captcha" value="{{.captcha}}" autocomplete="off">
+		<input id="captcha" class="gt-w-full" name="captcha" value="{{.captcha}}" autocomplete="off">
```
Comment (wxiaoguang) [inline templates/user/auth/captcha.tmpl]: I just realized that it is playing a CSS style overriding game ....

For example: by Fomantic UI: the `inline field` above means that the label is the same row as the input

https://fomantic-ui.com/collections/form.html#inline-field


So, by default, without the `inline`, it should be the UI you need. 


I do not think it is right to use "inline" outside, and change the "inline" behavior inside.

There are already many UI bugs caused by such unclear overridings.

**Comment by lunny** [comment]
I personally like the previous style. It's more beautiful to not align every line.

**Review by wxiaoguang — DISMISSED** [review]
Put a "request change" by https://github.com/go-gitea/gitea/pull/29432#discussion_r1503566005

Personally I think it should clearly define the HTML layout and styles, instead of overriding the styles to a different purpose.

If most people agree to override, feel free to dismiss this request change.

Thank you.

File: templates/user/auth/captcha.tmpl (hunk 1 above)
Comment (rafh) [reply templates/user/auth/captcha.tmpl]: I understand this perspective. I removed the inline class to allow the default behavior of the `div` to take over. 

On the topic of helper classes, we recently implemented tailwinds. We should update our contribution docs to reflect the usage of these classes? If tailwinds is the path forward, I'll open a PR to update these docs.

https://github.com/go-gitea/gitea/blob/4b8293aa094e725b372329a19da687a6d1550069/docs/content/contributing/guidelines-frontend.en-us.md?plain=1#L50

Comment (wxiaoguang) [reply templates/user/auth/captcha.tmpl]: I think it should be updated accordingly. At the moment the plan is to remove most Gitea's own helpers (`gt-`), use Tailwind (`tw-`) as much as possible.

**Review by wxiaoguang — COMMENTED** [review]
Thank you very much. Without the "inline" patches, the new approach looks better to me.

* It seems that the ".user.activate" form is still using "inline", should it be updated too?
* Why sometimes there is "md:tw-max-w-2xl tw-m-auto" but sometimes there isn't? Could there be some general (unified) framework approaches?

**Comment by rafh** [comment]
> * md:tw-max-w-2xl

This [line of CSS](https://github.com/go-gitea/gitea/pull/29432/files#diff-27543749149b34d4bb8459b8e24c992af84863721d6443a53d95ec0e43f39860L246
) is the reason for the use of `md:tw-max-w-2xl tw-m-auto`. To create a unified approach, I removed the CSS and updated the templates that used it. The PR grew, but things are now more consistent.

**Comment by wxiaoguang** [comment]
Thank you, it looks better. Actually, I am not a fan of fine-tuning everything, that's why I proposed to use some clearly defined CSS classes for these forms. But the current approach is not unacceptable ....

The last question: why `user.activate form` is left there? I think it also belongs to the "user sign in/out" pages.

**Review by wxiaoguang — APPROVED** [review]

**Comment by silverwind** [comment]
Sidenote: Please don't raise PR's from `main` branch for [reasons mentioned here](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/) and because it's a pain for reviewers to work with that branch as they need to rename their local branch.

**Review by silverwind — APPROVED** [review]

**Pull request merged**